pkg/trace/traceutil/azure_test.go

```diff
@@ -114,7 +135,27 @@ func mockAzureAppServiceMetadata() map[string]string {
 	return aasMetadata
 }
 
+func mockAzureFunctionAppMetadata() map[string]string {
+	aasMetadata = make(map[string]string)
+	aasMetadata["WEBSITE_SITE_NAME"] = "site-name-test"
+	aasMetadata["WEBSITE_OWNER_NAME"] = "00000000-0000-0000-0000-000000000000+apm-dotnet-EastUSwebspace-Linux"
+	aasMetadata["WEBSITE_RESOURCE_GROUP"] = "test-resource-group"
+	aasMetadata["WEBSITE_INSTANCE_ID"] = "1234abcd"
+	aasMetadata["COMPUTERNAME"] = "test-instance"
+	aasMetadata["WEBSITE_STACK"] = "NODE"
+	aasMetadata["WEBSITE_NODE_DEFAULT_VERSION"] = "~18"
+	aasMetadata["FUNCTIONS_WORKER_RUNTIME"] = "node"
+	aasMetadata["FUNCTIONS_EXTENSION_VERSION"] = "~4"
+
+	return aasMetadata
+}
```

Comment:
Why not just make this a global? Since it doesn't ever change, why rebuild it every time you need it?

Reply:
Also, not a huge deal here because this is a test, but in general, this method of defining a map can lead to unnecessary allocations, which are really slow in go.

There's two things you can do to avoid this. If you already know the entire contents of the map, define it as

```go
myMap := map[string]string{
    "key1": "value1",
    "key2": "value2",
    ... etc ...
}
```

Alternatively, you can tell `make` just how much memory to allocate for the map. There is an optional second argument to `make` which is the "capacity". For example

```go
myMap := make(map[string]string, 10)
myMap["key1"] = "value1"
myMap["key2"] = "value2"
... etc ...
```

I forget the exact details on map and slice implementation in go, but it's something like this. When one of these times is `make`'d, it is initially allocated with a very small number of slots, like 1 or 2 or something. Once the map/slice reaches its capacity, the go runtime will, I believe, double its capacity, in this case from 1 to 2 or from 2 to 4, etc. This means that for your above example, you'd be preforming something like 4 allocations.